bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/ClientLibraryImplTest.java

```diff
@@ -15,94 +15,126 @@
+
+        assertEquals("", clientLibrary.getInline());
     }
 
     @Test
-    public void getInline() {
-        this.cl.categories = this.categories;
-        this.cl.type = this.type;
+    void testEmptyGetInlineJs() throws Exception {
+        context.registerInjectActivateService(new MockHtmlLibraryManager(mock(com.adobe.granite.ui.clientlibs.ClientLibrary.class)));
+        context.registerInjectActivateService(new ClientLibraryAggregatorServiceImpl());
+        ClientLibrary clientLibrary = getClientLibraryUnderTestFromResource(CLIENT_LIB_JS);
 
-        when(this.clientLibraryAggregatorServiceMock.getClientLibOutput(this.categories, this.type))
-          .thenReturn("output inline");
+        assertEquals("", clientLibrary.getInline());
+    }
+
+    @Test
+    void testGetInlineCss() throws Exception {
+        HtmlLibraryManager htmlLibraryManager = context.registerInjectActivateService(mock(MockHtmlLibraryManager.class));
+        context.registerInjectActivateService(new ClientLibraryAggregatorServiceImpl());
+        HtmlLibrary library = mock(HtmlLibrary.class);
 
-        assertEquals("output inline", this.cl.getInline());
+        ClientLibrary clientLibrary = getClientLibraryUnderTestFromResource(CLIENT_LIB_CSS);
+        Resource clientLibResource = context.currentResource(TEST_APPS_ROOT + "/clientlib-base-amp/styles/index.css");
+
+        when(htmlLibraryManager.getLibraries(any(), any(), anyBoolean(), anyBoolean())).thenReturn(Arrays.asList(mock(com.adobe.granite.ui.clientlibs.ClientLibrary.class)));
+        when(htmlLibraryManager.getLibrary(any(), any())).thenReturn(library);
+        when(library.getInputStream(anyBoolean())).thenReturn(clientLibResource.adaptTo(InputStream.class));
+
+        String outputString = IOUtils.toString(clientLibResource.adaptTo(InputStream.class), StandardCharsets.UTF_8);
+        assertEquals(outputString, clientLibrary.getInline());
+    }
+
+    @Test
+    void testGetInlineJs() throws Exception {
+        HtmlLibraryManager htmlLibraryManager = context.registerInjectActivateService(mock(MockHtmlLibraryManager.class));
+        context.registerInjectActivateService(new ClientLibraryAggregatorServiceImpl());
+        HtmlLibrary library = mock(HtmlLibrary.class);
+
+        ClientLibrary clientLibrary = getClientLibraryUnderTestFromResource(CLIENT_LIB_JS);
+        Resource clientLibResource = context.currentResource(TEST_APPS_ROOT + "/clientlib-base-amp/scripts/index.js");
+
+        when(htmlLibraryManager.getLibraries(any(), any(), anyBoolean(), anyBoolean())).thenReturn(Arrays.asList(mock(com.adobe.granite.ui.clientlibs.ClientLibrary.class)));
+        when(htmlLibraryManager.getLibrary(any(), any())).thenReturn(library);
+        when(library.getInputStream(anyBoolean())).thenReturn(clientLibResource.adaptTo(InputStream.class));
+
+        String outputString = IOUtils.toString(clientLibResource.adaptTo(InputStream.class), StandardCharsets.UTF_8);
```

Comment:
same as above

Reply:
Updated the `outputString` to be a hardcoded string value.